### PR TITLE
Fix/create post

### PR DESCRIPTION
**Pull request by Lil-Noe**
This pull request includes several updates to the UI tests and various components to improve the handling of challenge data and descriptions. The most important changes involve modifying the challenge creation screen to fetch and display images and descriptions, and updating the home screen to fetch and display the current challenge details.

### Updates to UI Tests:
* [`__tests__/UI-tests/components/challenge.test.tsx`](diffhunk://#diff-da079f2863696d41f5cf82775a5e431454d52cf20b5d0b3111a38018a68bb178L1-R1): Modified the challenge data structure and added placeholder tests. [[1]](diffhunk://#diff-da079f2863696d41f5cf82775a5e431454d52cf20b5d0b3111a38018a68bb178L1-R1) [[2]](diffhunk://#diff-da079f2863696d41f5cf82775a5e431454d52cf20b5d0b3111a38018a68bb178L45-L53) [[3]](diffhunk://#diff-da079f2863696d41f5cf82775a5e431454d52cf20b5d0b3111a38018a68bb178R182-R188)
* [`__tests__/UI-tests/createChallenge.test.tsx`](diffhunk://#diff-9c5cbebe3cfaf1c58061456979ebb55fc225240fde3fa9f7ec50c80dec17261fR74-R77): Updated the test to reflect changes in the challenge form fields.
* [`__tests__/UI-tests/home/maximize.test.tsx`](diffhunk://#diff-6c25f5d97b3cd6447925c3ad43d5af641d1045a49b78fbde8a0863d8366793d6L67-R67): Updated the test to reflect changes in challenge data fields. [[1]](diffhunk://#diff-6c25f5d97b3cd6447925c3ad43d5af641d1045a49b78fbde8a0863d8366793d6L67-R67) [[2]](diffhunk://#diff-6c25f5d97b3cd6447925c3ad43d5af641d1045a49b78fbde8a0863d8366793d6L102)
* [`__tests__/UI-tests/map/MapScreen.test.tsx`](diffhunk://#diff-30cc182bb42fc5161b42d888aa38143ef04f33bab7eb065b643355c7b74b5f9fL1-R111): Refactored the test to include mock data and verify map rendering and location permissions.

### Changes to Challenge Creation Screen:
* [`app/screens/create/create_challenge.tsx`](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294L2-R2): Updated to fetch and display challenge images and descriptions, and modified the form fields for creating challenges. [[1]](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294L2-R2) [[2]](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294R18-R50) [[3]](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294R69-R105) [[4]](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294R131-R158) [[5]](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294R207-R224) [[6]](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294R233-R250) [[7]](diffhunk://#diff-a4a786d5b02a6a44b0e1e61d78c950a5828231f50937e5bc1fe5f5b687999294L208-R279)

### Updates to Home Screen:
* [`app/screens/home/home_screen.tsx`](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59L43-R50): Updated to fetch and display the current challenge details and sorted challenges by date. [[1]](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59L43-R50) [[2]](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59R63-R111) [[3]](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59L158-R187) [[4]](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59R198)

### Updates to Maximize Screen:
* [`app/screens/home/maximize_screen.tsx`](diffhunk://#diff-a8a944098359f50dc2fc9d15bc5295d0895d96be7143ba171a0b509c0d2fef71L77-R84): Modified to use the new challenge data fields and removed the description section. [[1]](diffhunk://#diff-a8a944098359f50dc2fc9d15bc5295d0895d96be7143ba171a0b509c0d2fef71L77-R84) [[2]](diffhunk://#diff-a8a944098359f50dc2fc9d15bc5295d0895d96be7143ba171a0b509c0d2fef71L195-L203) [[3]](diffhunk://#diff-a8a944098359f50dc2fc9d15bc5295d0895d96be7143ba171a0b509c0d2fef71L349-L357)

### CODE REVIEW

File: __tests__/View-tests/home/homeScreen.test.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React, { act } from "react";
 import { render, waitFor, fireEvent } from "@testing-library/react-native";
 import HomeScreen from "@/src/views/home/home_screen";
 import FirestoreCtrl from "@/src/models/firebase/FirestoreCtrl";
+import { get } from "react-native/Libraries/TurboModule/TurboModuleRegistry";
```
Comment (Roman-batut) [inline __tests__/View-tests/home/homeScreen.test.tsx:5]: This import is unused.

File: __tests__/Components/home/challenge.test.tsx [hunk 2]
```diff
@@ -106,7 +107,7 @@ describe("Challenge Component", () => {
   });
 
   it("navigates to Maximize screen when expand button is pressed", async () => {
-    render(
+    const { getByTestId } = render(
```
Comment (Roman-batut) [inline __tests__/Components/home/challenge.test.tsx:110]: `getByTestId` is unused in your test

**Review by Roman-batut — CHANGES_REQUESTED** [review]
Nice PR, I suggested some changements to fix some errors in the code, check them out ! I will be happy to approve the PR when these changements will be fixed

File: __tests__/Components/home/challenge.test.tsx [hunk 3]
```diff
@@ -118,6 +119,7 @@ describe("Challenge Component", () => {
     );
 
     // Open the detailed view
+
     await waitFor(() => {
       fireEvent.press(screen.getByTestId("add-a-comment"));
```
Comment (EduardoNeville) [inline __tests__/Components/home/challenge.test.tsx:124]: > `getByTestId` is unused in your test

Used here no?

File: __tests__/View-tests/map/mapScreen.test.tsx [hunk 4]
```diff
@@ -19,8 +20,12 @@ jest.mock("react-native-maps", () => {
 });
 
 describe("MapScreen UI Tests", () => {
-  const mockNavigation = { goBack: jest.fn() };
+  const mockNavigation = {
+    goBack: jest.fn(),
+    navigate: jest.fn(),
+  };
   const mockFirestoreCtrl = new FirestoreCtrl();
+  const mockDate = new Date();
```
Comment (EduardoNeville) [inline __tests__/View-tests/map/mapScreen.test.tsx:28]: When using new Date for a mock make sure to accept any date as an expected argument

File: app/components/home/Challenge.tsx [hunk 5]
```diff
@@ -46,6 +47,12 @@ export function Challenge({
     const fetchUser = async () => {
       try {
         const userData = await firestoreCtrl.getUser(challengeDB.uid);
+        if (
+          userData.image_id?.startsWith("http") ||
```
Comment (EduardoNeville) [inline app/components/home/Challenge.tsx:51]: Why use an insecure protocol?

File: app/src/models/firebase/FirestoreCtrl.ts [hunk 6]
```diff
@@ -35,14 +35,14 @@ export type DBUser = {
...
-  challenge_name: string;
-  description?: string;
+  caption: string;
   uid: string;
   image_id?: string;
   date?: Date;
   likes?: string[]; // User IDs
   location?: GeoPoint | null;
   group_id?: string;
+  challenge_description: string;
```
[the remark this answers is not in the document]
Comment (EduardoNeville) [reply app/src/models/firebase/FirestoreCtrl.ts:45]: agreed remove challenge_description no?

File: app/src/models/types/ChallengeBuilder.tsx
[the remark this answers is not in the document]
Comment (EduardoNeville) [reply app/src/models/types/ChallengeBuilder.tsx:1]: Remove ``models/ChallengeBuilder.tsx`` and keep  ``types/ChallengeBuilder.tsx``

File: app/src/models/types/ChallengeBuilder.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        : new GeoPoint(location.coords.latitude, location.coords.longitude);
+
+    const newChallenge: DBChallenge = {
+      caption: caption || "",
+      uid: user.uid,
+      image_id: image_id,
+      likes: likes || [],
+      location: locationFirebase,
+      challenge_description: challenge_description,
+      date: date,
```
[the remark this answers is not in the document]
Comment (EduardoNeville) [reply app/src/models/types/ChallengeBuilder.tsx:39]: We need to take either one or the other but not both

File: app/src/viewmodels/home/MaximizeScreenViewModel.tsx [hunk 8]
```diff
@@ -56,6 +57,23 @@ export function useMaximizeScreenViewModel(
     });
   }, [challenge, firestoreCtrl, currentUserId]);
 
+  // Fetch user data
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        if (
+          user.image_id?.startsWith("http") ||
+          user.image_id?.startsWith("https://")
```
Comment (EduardoNeville) [inline app/src/viewmodels/home/MaximizeScreenViewModel.tsx:66]: Same question here. What is the reason of this divide?

File: app/types/ChallengeBuilder.tsx [hunk 9]
```diff
@@ -26,11 +26,14 @@ export const buildChallenge = async (
     await firestoreCtrl.getName(challenge.uid);
 
     const challengeData: DBChallenge = {
-      challenge_name: challenge.challenge_name,
-      description: challenge.description,
+      caption: challenge.caption,
```
Comment (EduardoNeville) [inline app/types/ChallengeBuilder.tsx:29]: Caption and challenge Description?

File: app/types/ChallengeBuilder.tsx [hunk 10]
```diff
@@ -71,29 +74,21 @@ export const createChallenge = async (
...
     const newChallenge: DBChallenge = {
-      challenge_name: challenge_name,
-      description: description || "",
+      caption: caption || "",
       uid: user.uid,
       image_id: image_id,
       likes: likes || [],
       location: locationFirebase,
       group_id: group_id,
+      challenge_description: challenge_description,
```
Comment (EduardoNeville) [inline app/types/ChallengeBuilder.tsx:83]: Challenge Description as well as Caption?

File: app/types/ChallengeBuilder.tsx [hunk 11]
```diff
@@ -71,29 +74,21 @@ export const createChallenge = async (
...
 
-    if (image_id) {
-      const image_url = await firestoreCtrl.getImageUrl(image_id);
-      newChallenge.image_id = image_url;
-    }
-
-    if (date) {
-      newChallenge.date = date;
-    }
-
```
[the remark this answers is not in the document]
Comment (EduardoNeville) [reply app/types/ChallengeBuilder.tsx:91]: Yes, this is handled in #253

File: package.json [hunk 12]
```diff
@@ -10,7 +10,7 @@
     "web": "expo start --web",
     "test": "jest",
     "lint": "expo lint",
-    "format": "npx prettier --check ."
+    "format": "npx prettier --write ."
```
Comment (EduardoNeville) [inline package.json:13]: Good call we should implement this everywhere to not worry about formatting

**Review by EduardoNeville — CHANGES_REQUESTED** [review]
Some changes to discuss.All in all great PR

File: package.json
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply package.json:1]: The command to apply the format would be shorter and more intuitive.

File: app/src/models/firebase/FirestoreCtrl.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/models/firebase/FirestoreCtrl.ts:45]: This is the challenge_Title acthually (weird names conventions) to be able to filter posts if there are multiples titles exitsing on firebase (Christmas-linkes posts should not be there if the new Challenge title is about easter for example)

File: app/src/models/types/ChallengeBuilder.tsx
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/models/types/ChallengeBuilder.tsx:1]: Done !

File: app/src/models/types/ChallengeBuilder.tsx [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Used to create a Challenge and store it in Firestore DB
+ */
+export const createChallenge = async (
+  firestoreCtrl: FirestoreCtrl,
+  caption: string,
+  location: LocationObject | null,
+  challenge_description: string,
+  date?: Timestamp,
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/models/types/ChallengeBuilder.tsx:13]: It is in the other ChallengeBuilder file

File: app/src/models/types/ChallengeBuilder.tsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
+import FirestoreCtrl, { DBChallenge, DBUser } from "../firebase/FirestoreCtrl";
+import { GeoPoint, Timestamp } from "firebase/firestore";
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/models/types/ChallengeBuilder.tsx:2]: Done from the Firebase file

File: app/src/models/types/ChallengeBuilder.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/models/types/ChallengeBuilder.tsx:39]: Resolved as before

File: app/src/models/types/ChallengeBuilder.tsx [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+      uid: user.uid,
+      image_id: image_id,
+      likes: likes || [],
+      location: locationFirebase,
+      challenge_description: challenge_description,
+      date: date,
+    };
+
+    if (date) {
+      newChallenge.date = date;
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/models/types/ChallengeBuilder.tsx:43]: Same as before

File: app/src/viewmodels/camera/CameraViewModel.tsx [hunk 16]
```diff
@@ -41,7 +44,14 @@ export default function useCameraViewModel(
...
   // Challenge state
-  const [description, setDescription] = useState("");
+  const [caption, setCaption] = useState("");
+
+  const [descriptionTitle, setDescriptionTitle] =
+    useState<DBChallengeDescription>({
+      title: "Challenge Title",
+      description: "Challenge Description",
+      endDate: new Date(2024, 1, 1, 0, 0, 0, 0),
+    });
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/viewmodels/camera/CameraViewModel.tsx:54]: This is the Title Challenge that is supposed to be shared by all posts, so it needs to be there as we create the posts in this file.

File: app/src/viewmodels/camera/CameraViewModel.tsx [hunk 17]
```diff
@@ -96,17 +106,35 @@ export default function useCameraViewModel(
...
 
+  // Fetch the description title
+  useEffect(() => {
+    async function fetchDescriptionTitle() {
+      try {
+        const currentChallengeData =
+          await firestoreCtrl.getChallengeDescription();
+
+        setDescriptionTitle(currentChallengeData);
+        console.log("Description title: ", currentChallengeData.title);
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/viewmodels/camera/CameraViewModel.tsx:117]: Done thank you !

File: app/types/ChallengeBuilder.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/types/ChallengeBuilder.tsx:91]: I call this method in the CameraContainer now

File: app/src/views/home/maximize_screen.tsx [hunk 18]
```diff
@@ -65,7 +65,7 @@ export default function MaximizeScreen({
   return (
     <ThemedView style={styles.bigContainer}>
       <TopBar
-        title={postTitle}
+        title=""
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/views/home/maximize_screen.tsx:68]: It seems this should be fixed in the TopBar component.

File: __tests__/View-tests/map/mapScreen.test.tsx (hunk 4 above)
Comment (Lil-Noe) [reply __tests__/View-tests/map/mapScreen.test.tsx:28]: I get it thank you !

File: app/components/home/Challenge.tsx (hunk 5 above)
Comment (Lil-Noe) [reply app/components/home/Challenge.tsx:51]: The creation of the URL doesn't depend entirely on us but more on the firebase methods so it seems it should be there still.

File: app/src/viewmodels/home/MaximizeScreenViewModel.tsx (hunk 8 above)
Comment (Lil-Noe) [reply app/src/viewmodels/home/MaximizeScreenViewModel.tsx:66]: Same as previous

File: app/types/ChallengeBuilder.tsx (hunk 9 above)
Comment (Lil-Noe) [reply app/types/ChallengeBuilder.tsx:29]: challenge Description is here the title of the challenge which will be used to filter posts.

**Review by Roman-batut — APPROVED** [review]
You correctly implemented the requested changes. Bravo !
### LGTM

**Review by EduardoNeville — APPROVED** [review]

**Pull request merged**